### ISSUE TITLE
When using model.info() to calculate FLOPs, it seems that there is an error in the calculation.

### ISSUE THREAD

**Issue by 885225250** [ultralytics/ultralytics#22636]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I use yolov12-obb.yaml to improve the model. However, when I modify any layer, even if I change the simple conv function to nn.Identity(), the model's FLOPs will soar from 6.x to 20.x. What's the reason for this?

# YOLO12n backbone
backbone:

  - [-1, 1, Conv, [64, 3, 2]] # 0-P1/2
  
  #- [-1, 1, Conv, [128, 3, 2]] # 1-P2/4（The original）
  - [-1, 1, conv3,  [16]]（Replaced）

  - [-1, 2, C3k2, [256, False, 0.25]]
  - [-1, 1, Conv, [256, 3, 2]] # 3-P3/8
  - [-1, 2, C3k2, [512, False, 0.25]]
  - [-1, 1, Conv, [512, 3, 2]] # 5-P4/16
  - [-1, 4, A2C2f, [512, True, 4]]
  - [-1, 1, Conv, [1024, 3, 2]] # 7-P5/32
  - [-1, 4, A2C2f, [1024, True, 1]] # 8

conv3：

class conv3(nn.Module):

    def __init__(self, dim):
        super().__init__()

        self.idx=nn.Identity()

    def forward(self, x):

        x=self.idx(x)

        return x

The test train file：

from ultralytics import YOLO

model = YOLO('/mnt/c/project/ultralytics-main/RSAR/yolo12-obb.yaml') 
model.info()

**Result：**
The original：
WARNING ⚠️ no model scale passed. Assuming scale='n'.
YOLO12-obb summary: 287 layers, 2,640,885 parameters, 2,640,869 gradients, 6.7 GFLOPs

Replaced：
WARNING ⚠️ no model scale passed. Assuming scale='n'.
YOLO12-obb summary: 286 layers, 2,635,701 parameters, 2,635,685 gradients, 25.6 GFLOPs

**Comment by UltralyticsAssistant** [comment]
👋 Hello @885225250, thank you for opening this issue and for your interest in Ultralytics 🚀! This is an automated response to help you get support faster. An Ultralytics engineer will follow up with you here shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for quick references where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report (FLOPs discrepancy when modifying layers), please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate. In addition to what you shared, please include:
-Exact `ultralytics` and `torch` versions (`pip show ultralytics torch`)
-OS, CUDA/CUDNN, and GPU details
-The full minimal YAML (before/after) and the exact `conv3` module code
-How `conv3` is imported/registered so the YAML can locate it
-The exact script/commands used to run `model.info()` including any `imgsz` or other args
-Full console logs from `model.info()` output
-A tiny repro script that builds the model and calls `model.info()` on a clean environment

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the clear repro—this jump is expected because replacing `Conv(..., stride=2)` with an identity removes a downsampling step, so all subsequent layers run at 2× height and width (~4× more pixels), which turns ~6.7 GFLOPs into ~25–27 GFLOPs; `model.info()` computes FLOPs via `get_flops()` and scales by the model stride, see the [get_flops reference](https://docs.ultralytics.com/reference/utils/torch_utils/#get_flops) and [model.info()](https://docs.ultralytics.com/reference/nn/tasks/#info).

If you want “no-params” behavior but keep the FLOPs comparable, keep the downsample at that layer:

```yaml
# replace the stride-2 Conv with a stride-2 pool
- [-1, 1, nn.MaxPool2d, [2, 2]]
```

Quick check to confirm stride change:

```python
from ultralytics import YOLO
m0 = YOLO("original.yaml"); m1 = YOLO("modified.yaml")
print("strides:", m0.model.stride.tolist(), m1.model.stride.tolist())
```

If you still suspect a bug with an unchanged YAML, please attach the full YAML and the text output of `model.info(imgsz=640, detailed=True)` plus `yolo checks` (or `!yolo checks` in notebooks) so we can verify.
